### ISSUE TITLE
Builds failed upgrading from ExpoKit 34 to ExpoKit 36: ':expo-location:compileDebugJavaWithJavac', ':react-native-spinkit:compileDebugJavaWithJavac', ':react-native-gesture-handler:compileDebugJavaWithJavac', ':app:debugCompileClasspath', ':app:preDebugBuild'

### ISSUE THREAD

**Issue by Andrea-Arguello** [expo/expo#7481]
### Environment
```
Expo CLI 3.15.5 environment info:
    System:
      OS: Linux 5.3 Pop!_OS 18.04 LTS
      Shell: 5.4.2 - /usr/bin/zsh
    Binaries:
      Node: 10.19.0 - /usr/bin/node
      Yarn: 1.22.4 - /usr/bin/yarn
      npm: 6.13.4 - /usr/bin/npm
    npmPackages:
      expo: ^36.0.0 => 36.0.2 
      react: 16.9.0 => 16.9.0 
      react-native: https://github.com/expo/react-native/archive/sdk-36.0.1.tar.gz => 0.61.4 
      react-navigation: ^3.1.4 => 3.11.1
```

The app target is Android, with ExpoKit.

### Steps to Reproduce
Add the react-native-spinkit and react-native-gesture to a project. Update from SDK 34 to SDK 35 (which also caused this problem) to SDK 36.

### Expected Behavior
Being able to run ```./gradlew installDebug``` without any compilation errors.

### Actual Behavior
I am currently having trouble with the two libraries mentioned above. It has come to my attention that both of these libraries seem to be using AndroidX, which seems to be causing the problem, since when I try to run ```./gradlew installDebug``` it crashes with the following errors:

```python
1: Task failed with an exception.
* What went wrong:
Execution failed for task ':expo-location:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

2: Task failed with an exception.
* What went wrong:
Execution failed for task ':react-native-spinkit:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

3: Task failed with an exception.
* What went wrong:
Execution failed for task ':react-native-gesture-handler:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

4: Task failed with an exception.
* What went wrong:
Could not resolve all files for configuration ':app:debugCompileClasspath'.
> Could not find com.facebook.react:react-native:36.0.1.
  Required by:
      project :app
> Could not find host.exp.exponent:expoview:36.0.1.
  Required by:
      project :app
> Could not find com.facebook.react:react-native:36.0.1.
  Required by:
      project :app > project :react-native-pspdfkit
```

I tried to solve this by adding ```android.useAndroidX=true``` and ```android.enableJetifier=true``` to the gradle.properties, but this leads to a whole new problem which I don't know if it's because of my addition, or if it has to do with me configuring my update to ExpoKit 36 wrong. I get the following message:

```python
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:preDebugBuild'.
> Could not resolve all task dependencies for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.facebook.react:react-native:36.0.1.
         Required by:
                  project :app > project :unimodules-react-native-adapter
                  project :app > project :react-native-orientation-locker
                  project :app > project :react-native-device-info
                  project :app > project :rn-fetch-blob
                  project :app > project :react-native-fs
                  project :app > project :react-native-view-pdf
                  project :app > project :react-native-pspdfkit
                  project :app > project :react-native-gesture-handler
                  project :app > project :react-native-vector-icons
                  project :app > project :react-native-spinkit
```

Am I not supposed to add AndroidX to my project? Or is the latter problem the real issue? 

If needed, I may attach my configuration for updating to ExpoKit 36, or any other information that may help resolve this issue

**Comment by Andrea-Arguello** [comment]
I have changed in my android/app/build.gradle the following:
```diff
-api 'com.facebook.react:react-native:36.0.1'
+api 'com.facebook.react:react-native:0.61.4'
```
[The first one is what the documentation is telling me to do](https://docs.expo.io/versions/latest/expokit/expokit/#upgrading-expokit), so either there must be a linking error or it is a documentation bug.

However, I keep getting
```python
> Could not find host.exp.exponent:expoview:36.0.1.
  Required by:
      project :app
```
Which I guess has to do with the line on the android/app/build.gradle
``` implementation('host.exp.exponent:expoview:36.0.1@aar') {```
However, changing it to 36.0.0 doesn't make the error go away.

**Comment by Andrea-Arguello** [comment]
The solution to my last comment seems to have been solved by the solution in #4029
